pydm/tests/widgets/test_enum_combo_box.py

```diff
@@ -0,0 +1,330 @@
+    (False, True, True, False),
+
+    (False, False, True, True),
+    (False, False, True, False),
+
+    (False, True, False, True),
+    (False, True, False, False),
+
+    (False, False, False, True),
+    (False, False, False, False),
+])
+def test_check_enable_state(qtbot, signals, monkeypatch, connected, write_access, has_enum, is_app_read_only):
+    """
+    Test the tooltip generated depending on the channel connection, write access, whether the widget has enum strings,
+    and whether the app is read-only.
+
+    Expectations:
+    1. If the data channel is disconnected, the widget's tooltip will display "PV is disconnected"
+    2. If the data channel is connected, but it has no write access:
+        a. If the app is read-only, the tooltip will read  "Running PyDM on Read-Only mode."
+        b. If the app is not read-only, the tooltip will read "Access denied by Channel Access Security."
+    3. If the widget does not have any enum strings, the tooltip will display "Enums not available".
+
+    Parameters
+    ----------
+    qtbot : fixture
+        Window for widget testing
+    signals : fixture
+        The signals fixture, which provides access signals to be bound to the appropriate slots
+    monkeypatch : fixture
+        To override the default behavior of PyDMApplication.is_read_only()
+    connected : bool
+        True if the channel is connected; False otherwise
+    write_access : bool
+        True if the widget has write access to the channel; False otherwise
+    has_enum: bool
+        True if the widget has enum strings populated; False if the widget contains no enum strings (empty of choices)
+    is_app_read_only : bool
+        True if the PyDM app is read-only; False otherwise
+    """
+    pydm_enumcombobox = PyDMEnumComboBox()
+    qtbot.addWidget(pydm_enumcombobox)
+
+    signals.write_access_signal[bool].connect(pydm_enumcombobox.writeAccessChanged)
+    signals.write_access_signal[bool].emit(write_access)
+
+    signals.connection_state_signal[bool].connect(pydm_enumcombobox.connectionStateChanged)
+    signals.connection_state_signal[bool].emit(connected)
+
+    signals.enum_strings_signal[tuple].connect(pydm_enumcombobox.enumStringsChanged)
+    signals.enum_strings_signal[tuple].emit(("START", "STOP", "PAUSE"))
+
+    monkeypatch.setattr(PyDMApplication, 'is_read_only', lambda *args: is_app_read_only)
+
+    original_tooltip = "Original Tooltip"
+    pydm_enumcombobox.setToolTip(original_tooltip)
+    pydm_enumcombobox.check_enable_state()
+
+    actual_tooltip = pydm_enumcombobox.toolTip()
+    if not pydm_enumcombobox._connected:
+        assert "PV is disconnected." in actual_tooltip
+    elif not write_access:
+        if is_pydm_app() and pydm_enumcombobox.app.is_read_only():
+            assert "Running PyDM on Read-Only mode." in actual_tooltip
+        else:
+            assert "Access denied by Channel Access Security." in actual_tooltip
+    elif not pydm_enumcombobox._has_enums:
```

Comment:
Is this condition ever being tested since you are always emitting the enum_strings?
You should have a conditional there looking for your parameter and emitting it or not, right?

Reply:
Right. This is missing a check for the parametrized has_enums.